index.js

```diff
@@ -1,107 +1,55 @@
 const core = require("@actions/core");
-const GitHubAPI = require("./githubapi");
-const OpenAIAPI = require("./openaiapi");
 
-const isFileToReview = (filename, fileExtensions, excludePaths) => {
-    if (fileExtensions) {
-        const extensions = fileExtensions.split(",").map((ext) => ext.trim());
-        if (!extensions.some((ext) => filename.endsWith(ext))) {
-            return false;
-        }
-    }
-
-    if (excludePaths) {
-        const paths = excludePaths.split(",").map((path) => path.trim());
-        if (paths.some((path) => filename.startsWith(path))) {
-            return false;
-        }
-    }
-
-    return true;
-}
+const GitHubAPI = require("./githubapi.js");
+const OpenAIAgent = require("./openai_agent.js");
 
-const getFilteredChangedFiles = (changedFiles, fileExtensions, excludePaths) => {
-    let filteredFiles = changedFiles;
-    return filteredFiles.filter((file) => isFileToReview(file.filename, fileExtensions, excludePaths));
-};
+const main = async () => {
+    const getFilteredChangedFiles = (changedFiles, includeExtensions, excludeExtensions, includePaths, excludePaths) => {
+        const stringToArray = (inputString) => inputString.split(',').map(item => item.trim().replace(/\\/g, '/')).filter(Boolean);
```

Comment:
The variable `githubToken` is used but never declared or initialized in the refactored code, resulting in a potential runtime error when attempting to instantiate `GitHubAPI` with `githubToken`.